File: 03-1-Form1-app/app/src/main/java/com/example/form1/MainActivity.java
package com.example.form1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    Button proceedBt;
    Button cancelBt;
    EditText firstNameET;
    TextView statusTV;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //
        proceedBt = (Button) findViewById(R.id.proceedButton);
        cancelBt = (Button) findViewById(R.id.cancelButton);
        firstNameET = (EditText) findViewById(R.id.firstName);
        statusTV = (TextView) findViewById(R.id.statusTextView);

        cancelBt.setOnClickListener(view -> {
            firstNameET.setText("");
        });

        proceedBt.setOnClickListener(view -> {
            String name = firstNameET.getText().toString();
            if (name.equals("")) {
                firstNameET.setError("Fill it!");
                return;
            }

            statusTV.setText(R.string.form_data_saved_msg);
        });
    }
}

File: 03-1-Form1-app-kt1/app/src/main/java/com/example/form1_app_kt1/MainActivity.kt
package com.example.form1_app_kt1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val proceedBt = findViewById<Button>(R.id.proceedButton)
        val cancelBt = findViewById<Button>(R.id.cancelButton)
        val firstNameET = findViewById<EditText>(R.id.firstName)
        val statusTV = findViewById<TextView>(R.id.statusTextView)

        cancelBt.setOnClickListener {
            firstNameET.setText("")
            statusTV.setText("")
        }

        proceedBt.setOnClickListener {
            val name = firstNameET.text.toString()
            if (name.isEmpty()) {
                // firstNameET.error = "Fill it!"
                firstNameET.error = getString(R.string.fill_it)
                return@setOnClickListener
            }

            statusTV.setText(R.string.form_data_saved_msg)
        }
    }
}